myhpi/tenca_django/views.py

```diff
@@ -0,0 +1,281 @@
+        return super().get_context_data(**kwargs)
+
+    def form_valid(self, form):
+        try:
+            new_list = connection.add_list(form.cleaned_data["list_name"], self.request.user.email)
+            messages.success(
+                self.request,
+                _("The list {list} has been created successfully.").format(
+                    list=new_list.fqdn_listname
+                ),
+            )
+            return redirect("tenca_django:tenca_dashboard")
+        except Exception as e:
+            messages.error(self.request, _("An error occurred: {error}").format(error=e))
+            return self.render_to_response(self.get_context_data(form=form))
+
+
+class TencaSubscriptionView(FormView):
+    form_class = TencaSubscriptionForm
+    template_name = "tenca_django/manage_subscription.html"
+
+    def setup(self, request, *args, **kwargs):
+        super().setup(request, *args, **kwargs)
+        self.mailing_list = connection.get_list_by_hash_id(kwargs["hash_id"])
+        if not self.mailing_list:
+            raise Http404
+
+    def get_context_data(self, **kwargs):
+        kwargs.setdefault("list_name", self.mailing_list.fqdn_listname)
+        return super().get_context_data(**kwargs)
+
+    def get_initial(self):
+        initial_mail = None
+        if self.request.user.is_authenticated:
+            # try:
+            #     initial_mail = next(
+            #         test_mail for test_mail in alternative_emails(self.request.user.email)
+            #         if self.mailing_list.is_member(test_mail)
+            #     )
+            # except StopIteration:
+            #     initial_mail = self.request.user.email
+            initial_mail = self.request.user.email
+        return {"email": initial_mail}
+
+    def form_valid(self, form):
+        try:
+            joined, token = self.mailing_list.toggle_membership(form.cleaned_data["email"])
+            messages.success(
+                self.request, _("The change has been requested. Please check your mails.")
+            )
+            return redirect(reverse("tenca_django:tenca_dashboard"))
+        except Exception as e:
+            messages.error(self.request, _("An error occurred: {error}").format(error=e))
+            return self.render_to_response(self.get_context_data(form=form))
+
+
+class TencaListAdminView(LoginRequiredMixin, TencaListAdminMixin, FormView):
+    template_name = "tenca_django/manage_list.html"
+
+    def get_form(self, form_class=None):
+        return TencaListOptionsForm(self.request.POST or None, mailing_list=self.mailing_list)
+
+    def get_context_data(self, **kwargs):
+        kwargs.setdefault("mailing_list", self.mailing_list)
+        kwargs.setdefault("listname", self.mailing_list.fqdn_listname)
+        # kwargs.setdefault("invite_link", urllib.parse.urljoin("https://" + settings.PAGE_URL, reverse("tenca_django:tenca_manage_subscription", kwargs=dict(hash_id=self.mailing_list.hash_id))))
```

Comment:
why is this commented out?

Reply:
Already commented out in the 1327 version

Reply:
just saw that there is also another line below that sets the invite_link kwarg. So i think this line can be removed